provider/datagen/src/transform/cldr/time_zones/convert.rs

```diff
@@ -368,3 +437,57 @@ fn iterate_zone_format_for_time_zone_id(
             )
         })
 }
+
+fn iterate_metazone_period(
```

Comment:
> nit | suggestion

I would probably call this `metazone_periods_iter` because it produces an iterator. This is just a matter of opinion, but I feel that I've seen this type of naming convention used more commonly. 

For example, `into_iter()` could have been called `iterate()`, but it is `into_iter()` instead.